app/main.py

```diff
@@ -1,3 +1,14 @@
-def cache(func):
-    # Write your code here
-    pass
+
+def cache(func: callable) -> any:
+    results = {}
+
+    def inner(*args) -> any:
+        if args in results.keys():
+            print("Getting from cache")
+            return results[args]
+        else:
+            result = func(*args)
+            results.update({args: result})
+            print("Calculating new result")
+            return result
```

Comment:
You can use only one return after `if`/`else` statements. For this purpose you need to do like this:
`results[args] = func(*args)`, and that's it.
There are 2 benefits in such approach: 
- You will need only one `return`
- Code will be written in more "pythonic" way